### DISCUSSION TITLE
azure devops - deployment mode

### DISCUSSION THREAD

**Discussion by mgajownik** [AzureCommunityPL/group#418] [Pomoc]
Cześć 

tworze IaC sieć z subnetami,, nsg z kilkoma regułami, vmki prosty template yaml z incrementa mode. no i fajnie poszedł deployment zrobiony.

dodałem kolejny job w yamlu do dodatkowych zasobów i po puszczeniu pipeline wywalił mi sie na pierwszym kroku z komunikatem, ze nie moze usunąć zasobu (subnetu) bo jest używany przez inny zasób. problem w tym ze ja go nie chciałem nawet zmianiać tylko dodać kolejny zasób ale w tym samym pipeline.

ogólnie jak robicie IaC wiele sieci, peeringi pomiędzy nimi

**Comment by maciej-skorupka** [comment]
Mówisz o ARMowych definicjach IaaC osadzonych w yaml pipelinu, tak? Jeśli jednak chodzi o definicje w yamlach, to wrzuć jakiś przykład o co chodzi, bo szczerze mówiąc nie kumam w takim przypadku. Zasadniczo definicje ARMowe nie powinny usuwać nagle same z siebie. 

Co do ogólnego podejścia, to jest mnóstwo sposobów, ale dookoła ARMa to miałbym repo z plikami ARM i pipelinem, który triggeruje się na zmianach w tych plikach. Pipeline odpala az cli (coś w stylu tego https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-to-resource-group?tabs=azure-cli#deployment-commands) po wejściu do katalogu w którym była zmiana (wykryte git diffem).
